### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by FlorianLudwig** [schildbach/public-transport-enabler#51]
The project is missing a readme, documentation and/or examples.

**Comment by ulich** [comment]
+1

Especially the legal information is something that is important imho. Is it ok to use the APIs of the public transport companies just by using your library (if yes, which)? I imagine that you need to get some permission of them, right? @schildbach What were your experiences with Öffi?

**Comment by jcharaoui** [comment]
I'd like to add support for my region (Canada, Quebec) and it looks easy enough but I'm a bit confused with the tests as the implementations vary a bit from region to region. Some quick guidelines to writing tests and an example would be a great help.

**Comment by grote** [comment]
Hi @jcharaoui, the information in grote/Transportr#144 might help you to get started. You might also be able to team up with @baldurmen.

**Comment by jcharaoui** [comment]
Yeah, we're already teamed up, and we're in the same situation. I've also looked at the pull request. My confusion stems from the fact that not all providers implement the same tests routines, and it's not clear to me what values should be provided to each test function call (valid or invalid data? data formatting, etc.).

**Comment by grote** [comment]
Great to hear that you teamed up already! :)

The tests have been written by different people and can always be improved. The more tests and the more code/API coverage, the better I'd say. Tests for Navitia providers currently in the public transport enabler have obviously been good enough for @schildbach, so this is the lower barrier you need to meet ;)

**Comment by baldurmen** [comment]
@jcharaoui  check this out: https://github.com/schildbach/public-transport-enabler/pull/76

The job is actually done, but it should answer your questions about tests

**Comment by grote** [comment]
Documentation was added with #73. Can we close this issue then?
